bio/cooltools/eigs_cis/wrapper.py

```diff
@@ -0,0 +1,44 @@
+__author__ = "Ilya Flyamer"
+__copyright__ = "Copyright 2022, Ilya Flyamer"
+__email__ = "flyamer@gmail.com"
+__license__ = "MIT"
+
+from snakemake.shell import shell
+
+## Extract arguments
+view = snakemake.params.get("view", "")
+if view:
+    view = f"--view {view}"
+track = snakemake.input.get("track", "")
+track_col_name = snakemake.params.get("track_col_name", "")
+if track and track_col_name:
+    track = f"--phasing-track {track}::{track_col_name}"
+elif track:
+    track = f"--phasing-track {track}"
+else:
+    track = ""
+
+extra = snakemake.params.get("extra", "")
+log = snakemake.log_fmt_shell(stdout=True, stderr=True)
+
+if "--bigwig" in extra:
+    mv_bigwig = " && mv {snakemake.output}.cis.bw {snakemake.output}.bw"
+else:
+    mv_bigwig = ""
+
+resolution = snakemake.params.get(
+    "resolution", snakemake.wildcards.get("resolution", 0)
+)
+if not resolution:
+    raise ValueError("Please specify resolution either as a wildcard or as a parameter")
+
+shell(
+    "(cooltools eigs-cis"
+    " {snakemake.input.cooler}::resolutions/{resolution} "
+    " {track}"
+    " {view} "
+    " {extra} "
+    " -o {snakemake.output[0]} && "
+    " mv {snakemake.output}.cis.vecs.tsv {snakemake.output} && "
+    " mv {snakemake.output}.cis.lam.txt {snakemake.output}.lam.txt {mv_bigwig}) {log}"
+)
```

Comment:
All output files of a rule should be specified under `output` and, if output specified, just set `--bigwig` automatically. Something like:

```suggestion
bigwig = snakemake.output.get("bigwig","")
if bigwig:
    bigwig = f"--bigwig {bigwig}"

resolution = snakemake.params.get(
    "resolution", snakemake.wildcards.get("resolution")
)
assert resolution, "Please specify resolution either as a `wildcard` or as a `parameter`"

with tempfile.TemporaryDirectory() as tmpdir:
    shell(
        "cooltools eigs-cis"
        " {snakemake.input.cooler}::resolutions/{resolution} "
        " {track}"
        " {view} "
        " {bigwig}"
        " {extra} "
        " -o {tmpdir}/out"
        " {log}"
    )

    shell("mv {tmpdir}/out.cis.vecs.tsv {snakemake.output.vecs}")
    shell("mv {tmpdir}/out.cis.lam.txt {snakemake.output.lam}")
    if bigwig:
        shell("mv {tmpdir}/out.cis.bw {snakemake.output.bigwig}")
```

Reply:
Done, thanks!